packages/dnd-timeline/src/hooks/useItem.tsx

```diff
@@ -112,22 +122,25 @@ export default function useItem(props: UseItemProps) {
 	useLayoutEffect(() => {
 		if (!dragDirection) return;
 
+		const calculateNewSpan = props.calcNewSan || defaultCalculateNewSpan;
+
 		const pointermoveHandler = (event: PointerEvent) => {
 			if (!dragStartX.current || !draggableProps.node.current) return;
 
 			const dragDeltaX =
 				(event.clientX - dragStartX.current) * (direction === "rtl" ? -1 : 1);
 
+			const newSpan = calculateNewSpan({
+				span: props.span,
+				direction: dragDirection,
+				deltaX: dragDeltaX,
+			});
+			const newWidth = valueToPixels(newSpan.end - newSpan.start);
+			draggableProps.node.current.style.width = `${newWidth}px`;
```

Comment:
For your particular use case, I suggest you to try and use the existing `getValueFromScreenX` function, which already implements "snap-to-grid" behavior.  
Maybe calculate the delta.  
Or maybe even export `snapValueToRangeGrid` function from `useTimeline` and use it here?